### PR TITLE
CM-420: developed new upload flow with validation

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://openimis.atlassian.net/browse/CM-420

### CODE REVIEW

**Review by delcroip — CHANGES_REQUESTED** [review]
Please add unit test for the service (direct or through API)

File: social_protection/services.py [hunk 1]
```diff
@@ -163,7 +243,74 @@ def _trigger_workflow(self,
...
+
+    def __calculate_percentage_of_invalid_items(self, upload_id):
+        total_items = len(self.__fetch_summary_of_broken_items(upload_id)) + len(
+            self.__fetch_summary_of_valid_items(upload_id))
+
+        if total_items == 0:
+            percentage_of_invalid_items = 0
+        else:
+            invalid_items = len(self.__fetch_summary_of_broken_items(upload_id))
+            percentage_of_invalid_items = (invalid_items / total_items) * 100
```
Comment (dborowiecki) [inline social_protection/services.py:312]: I think that both items are integers and in that case output will always be 0.

File: social_protection/services.py [hunk 2]
```diff
@@ -163,7 +243,74 @@ def _trigger_workflow(self,
...
+        ).values_list('uuid', flat=True))
+
+    def __calculate_percentage_of_invalid_items(self, upload_id):
+        total_items = len(self.__fetch_summary_of_broken_items(upload_id)) + len(
+            self.__fetch_summary_of_valid_items(upload_id))
+
+        if total_items == 0:
+            percentage_of_invalid_items = 0
+        else:
+            invalid_items = len(self.__fetch_summary_of_broken_items(upload_id))
```
Comment (dborowiecki) [inline social_protection/services.py:311]: Your calling `.__fetch_summary_of_broken_items` twice in this method. Could you do it once in order to improve performance?

**Review by dborowiecki — CHANGES_REQUESTED** [review]
I left two comments. Also please add tests in order to enhance coverage and pass quality gateway

**Comment by sniedzielski** [comment]
I added some basic test to check if the most important services works without raising any errors. I'll extend those tests later while developing another features.

**Review by dborowiecki — APPROVED** [review]

**Pull request merged**